scipy/io/tests/test_mmio.py

```diff
@@ -387,6 +387,20 @@ def test_sparse_formats(self):
                 result = mmread(fn).todense()
                 assert_array_almost_equal(result, expected)
 
+    def test_precision(self):
+        for precision in range(1, 8):
+            # construct sparse matrix with pi as last main diagonal entry
+            n = 10**precision + 1
+            A = scipy.sparse.lil_matrix((n, n))
+            A[n-1, n-1] = pi
+            # write matrix with low precision and read again
+            mmwrite(self.fn, A, precision=precision)
+            A = scipy.io.mmread(self.fn)
+            # check for right entries in matrix
+            assert_array_equal(A.row, [n-1])
+            assert_array_equal(A.col, [n-1])
+            assert_array_almost_equal(A.data, [pi], decimal=precision-1)
```

Comment:
Regarding the two options in https://github.com/scipy/scipy/pull/4865/files#r35975354, I'd much prefer ``fmt='%.4f'`` to ``fmt=%%.%if``. The latter is not really understandable imho.

Regarding backwards incompatibilities, it's slightly fuzzy what that means here. Only a change if you write+read, or a change in the actual output file (which is then read in in some non-Scipy way)? Technically it should be the latter I think. However for improving the precision I think it's justified to break backwards compat here either way. I haven't looked at this in too much detail, but my first impression is that @jor- has a point here with wanting to switch to ``%f``:

    In [1]: '%.5g' % 1.00001  # I'd hope that his works out of the box, not with a new fmt keyword
    Out[1]: '1'

    In [2]: '%.5e' % 1.00001   # This looks OK
    Out[2]: '1.00001e+00'

    In [3]: '%.5f' % 1.00001   # This looks OK as well.
    Out[3]: '1.00001'